libraries/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -42,6 +42,47 @@ export interface IRushProjectJson {
   operationSettings?: IOperationSettings[];
 }
 
+export interface IRushPhaseSharding {
+  /**
+   * The number of shards to create.
+   */
+  count: number;
+
+  /**
+   * The format of the argument to pass to the command to indicate the shard index and count.
+   * The default value is `--shard={shardIndex}/{shardCount}`.
+   */
+  shardArgumentFormat?: string;
+
+  /**
+   * An optional argument to pass to the command to indicate the output folder for the shard.
+   * The default argument is `--shard-output-folder=.rush/operations/{phaseName}/shards/{shardIndex}`.
+   */
+  outputFolderArgument?: {
+    /**
+     * The format of the parent folder name to output shards to. Child folders will be created for each
+     *  shard index, ie `{parentFolder}/1`, `{parentFolder}/2`. The default value is `.rush/operations/{phaseName}/shards`.
+     */
+    parentFolderName?: string;
+
+    /**
+     * The name of the argument to pass to the command to indicate the output folder for the shard.
+     * The default value is `--shard-output-folder`. Does not accept templating.
+     */
+    parameterLongName: string;
+  };
+
+  /**
+   * Configuration for the collator operation that combines the shards.
+   */
+  shardScriptConfiguration?: {
+    /**
+     * What should happen if the "_phase:<name>:shard" script is not defined in a project's package.json scripts field. Default is "error".
+     */
+    missingScriptBehavior?: 'silent' | 'log' | 'error';
+  };
```

Comment:
Why not just use the operation's `missingScriptBehavior`?

Reply:
Hmm, I think this shouldn't be configurable in the first place. Collation is optional but sharding should be required. This was from a previous iteration where the operation referenced the shard and not the collation. I'll remove this.

Reply:
I'll do a followup PR to align sharding with the new weighting, as I'd like to add weights to individual shards in my CI for jest tests.

Reply:
Do you want to do this before we publish? Will that be a breaking change if this is published first?

Reply:
It wouldn't be, bit of chaos in the comments here. I ended up getting the weights to work with sharding - it wasn't quite as bad as I thought it would be. Added in [791097f](https://github.com/microsoft/rushstack/pull/4652/commits/791097f8432104666a3d44d05e27092f2eae1a6b)